src/FsAutoComplete/LspServers/AdaptiveServerState.fs

```diff
@@ -37,6 +38,68 @@ open FsAutoComplete.Lsp
 open FsAutoComplete.Lsp.Helpers
 
 
+/// <summary>Handle tracking in-flight ServerProgressReport and allow cancellation of actions if a client decides to.</summary>
+type ServerProgressLookup() =
```

Comment:
A class to help track `ServerProgressReport` to easily cancel them. Hopefully comments help here.

Reply:
Also, what's a better name for this?